### PR TITLE
@feature/86ep3264f/Select-All-ThatApply_validation ft creating custon…

### PR DESCRIPTION
**Pull request by acostales08**
…Errorhandling hook

### CODE REVIEW

File: apps/simulator/src/components/blocks/RegularQuestions/SATA/RegularSATAQuestionaire.tsx [hunk 1]
```diff
@@ -1,58 +1,115 @@
...
+import { ControlledCheckbox } from '@/components/Checkbox';
+import {
+  useForm,
+  useFieldArray,
+  useFormState,
+  FormProvider
+} from 'react-hook-form';
+import { RegularSATAValidationAtom } from '@/core/schema/useAtomic';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { RegularSATAValidationType, RegSATASchema } from '@/core/schema/regularSATA/validation';
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/RegularQuestions/SATA/RegularSATAQuestionaire.tsx:15]: Is this being used? if not can we remove unused imports.

File: apps/simulator/src/components/blocks/RegularQuestions/SATA/RegularSATAQuestionaire.tsx [hunk 2]
```diff
@@ -1,58 +1,115 @@
...
+    <div className='h-full px-10 py-5'>
+      <Paper elevation={3}>
+        <FormProvider {...form}>
+          {itemselection?.length > 0 && itemselection.map((item: datatypes.CalcItemSelectValues, itemIndex) => (
+            <div key={itemIndex} className="py-8 px-16">
+              <p className="p-2 py-4">{item.question}</p>
+                <div className="">
+                {contents.answerUI?.length > 0 && contents.answerUI.map((answerUImap, answerUIidx) => {
+                  return(
+                    <>
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/RegularQuestions/SATA/RegularSATAQuestionaire.tsx:77]: React fragment here is useless.

File: apps/simulator/src/core/utils/contents.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    try {
+        if (choices === undefined || choices === null) {
+            console.error("Choices is undefined or null");
+            return null;
+        }
+
+        const parsedChoices = JSON.parse(choices);
+        return parsedChoices;
+    } catch (error) {
+        if (error instanceof Error) {
```
Comment (JMSevilla) [inline apps/simulator/src/core/utils/contents.ts:11]: Creating another condition for this error handler is unneccessary you can just have 1 console log and that's it.

File: apps/simulator/src/core/utils/contents.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+            console.error("An error occurred while parsing choices:", error.message);
+        } else {
+            console.error("An unknown error occurred while parsing choices");
+        }
+        return null;
```
Comment (JMSevilla) [inline apps/simulator/src/core/utils/contents.ts:16]: Remove return null

**Review by JMSevilla — CHANGES_REQUESTED** [review]

**Review by JMSevilla — APPROVED** [review]

**Review by JezChristian — APPROVED** [review]

**Review by DropitupQwerty — APPROVED** [review]

**Pull request merged**